src/core/Akka/Actor/ActorCell.Children.cs

```diff
@@ -233,7 +239,7 @@ private void ResumeChildren(Exception causedByFailure, IActorRef perpetrator)
             foreach (var stats in ChildrenContainer.Stats)
             {
                 var child = stats.Child;
-                var cause = child.Equals(perpetrator) ? causedByFailure : null;
+                var cause = (perpetrator != null && child.Equals(perpetrator)) ? causedByFailure : null;
```

Comment:
found an edge case where the `perpetrator` can be null which can cause a `NullReferenceException` to be thrown inside the `Equals` method. Figured it would be better to add the null check here rather than modify the `Equals` method.